src/test/ui/arbitrary_self_types_lifetime.rs

```diff
@@ -0,0 +1,70 @@
+#![feature(arbitrary_self_types)]
```

Comment:
Please move this file into some directory which is related to the feature. Test files should not be free-floating like this.

Reply:
The majority of related tests are in `run-pass/self`: https://github.com/rust-lang/rust/tree/master/src/test/run-pass/self. (There seems to be no folder dedicated to `arbitrary_self_types` feature)

Reply:
Let's put it in `ui/self` then and then move the `run-pass/self` tests over to `ui/self` later.

Reply:
#60950